bundler/lib/bundler/compact_index_client/updater.rb

```diff
@@ -49,16 +50,16 @@ def update(local_path, remote_path, retrying = nil)
 
         content = response.body
 
-        etag = (response["ETag"] || "").gsub(%r{\AW/}, "")
+        remote_sha256_digest = sha256_from_digest(response["Digest"]) || sha256_from_x_checksum(response["X-Checksum-Sha256"]) || ""
         correct_response = SharedHelpers.filesystem_access(local_temp_path) do
           if response.is_a?(Net::HTTPPartialContent) && local_temp_path.size.nonzero?
             local_temp_path.open("a") {|f| f << slice_body(content, 1..-1) }
 
-            etag_for(local_temp_path) == etag
+            sha256_digest_for_file(local_temp_path) == remote_sha256_digest
           else
             local_temp_path.open("wb") {|f| f << content }
 
-            etag.length.zero? || etag_for(local_temp_path) == etag
+            remote_sha256_digest.length.zero? || sha256_digest_for_file(local_temp_path) == remote_sha256_digest
```

Comment:
Alright, I'm convinced that there's very little downside to this. Resolving this thread.

Reply:
It would be cool to make a comment here that mentions that we're protecting against very rare things because we do so many requests that bytes that are transferred wrong of bits flipped on disk actually happen almost daily at this level of traffic.

Reply:
it isn't just that, it's also making sure the range request + the file on disk add together to the expected file (e.g. no earlier bytes than the requested range changed)

Reply:
In this case specifically, the http response can't be anything other than the full file, so checking the digest is just avoiding bit flips. In the other case though, you're right. @indirect clarified for me that the compact index versions file is recreated regularly and is not append only. That helped explain the partial response behavior.

This check that is highlighted here is truly just for making sure the file made it down correctly, since the entire response body is overwritten to the file.